src/graphql/schema/collection/createCollectionGqlType.ts

```diff
@@ -37,6 +37,7 @@ export default function createCollectionGqlType<TValue> (
         primaryKey && [options.nodeIdFieldName, {
           description: 'A globally unique identifier. Can be used in various places throughout the system to identify this single value.',
           type: new GraphQLNonNull(GraphQLID),
+          externalFieldNameDependencies: primaryKey._keyTypeFields.map(([fieldName, field]) => field.externalFieldName),
```

Comment:
`primaryKey._keyTypeFields` depends on an internal implementation detail, right? Any reason why we couldn’t just do: `externalKeyDependencies: [primaryKey]`. This would also work for some relations.